docs/pages/access-controls/idps/saml-reference.mdx

```diff
@@ -0,0 +1,129 @@
+
+This is the URL where the identity provider XML metadata can be retrieved from.
+Service providers will require information from this file to establish the
+trust between the service provider and identity provider.
+
+### Single Sign-On Service URL
+
+`https://<proxy-address>/enterprise/saml-idp/sso`
+
+This is the URL that service providers will use to exchange assertions with the
+identity provider. This URL supports the following bindings:
+
+- `HTTP-Redirect`
+- `HTTP-POST`
+
+### Identity provider initiated SSO URL
+
+`https://<proxy-address>/enterprise/saml-idp/login/<sp-name>`
+
+This URL triggers identity provider initiated SSO.
+
+## Teleport SAML service provider object
+
+Below is a service provider object. These objects must be added to Teleport in
+order for Teleport to trust external service providers.
+
+```yaml
+kind: saml_idp_service_provider
+metadata:
+  # The friendly name of the service provider. This is used to manage the
+  # service provider as well as in identity provider initiated SSO.
+  name: saml-service-provider
+spec:
+  # The entity_descriptor is the service provider XML.
+  entity_descriptor: |
+    <md:EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata"...
+version: v1
+```
+
+## Assertions
+
+The assertions currently provided by Teleport's SAML identity provider are listed below.
+
+| Friendly name          | Full name                           | Description               | Format                                            |
+|------------------------|-------------------------------------|---------------------------|---------------------------------------------------|
+| `uid`                  | `urn:oid:0.9.2342.19200300.100.1.1` | The user ID from Teleport | `urn:oasis:names:tc:SAML:2.0:attrname-format:uri` |
+| `eduPersonAffiliation` | `urn:oid:1.3.6.1.4.1.5923.1.1.1.1`  | The user's Teleport roles | `urn:oasis:names:tc:SAML:2.0:attrname-format:uri` |
+
+## Troubleshooting
+
+### `Bad Request` when logging into an external application
+
+If the Teleport server returns a `Bad Request` the service provider metadata may not
+have been added properly to Teleport and, as a result, Teleport is unable to find the
+service provider. You can verify this by looking for a log entry in Teleport's logs:
+
+```code
+2023-03-01T10:32:34-05:00 INFO [SAMLIDP]   failed to validate request: cannot find service provider https://samltest.id/saml/sp: could not find service provider https://samltest.id/saml/sp pid:31036.1 saml@v0.4.12/identity_provider.go:232
+```
+
+### `Not Found` when logging into an external application
+
+If the Teleport server returns a `Not Found`, make sure that none of the roles belonging
+to your user have SAML IdP access explicitly disabled. In the `options` section of each
+of the user roles, look for the `idp` section for the `saml` access to be disabled.
+
+```yaml
+...
+options:
+    ...
+    idp:
+      saml:
+        enabled: false
```

Comment:
Can we have a separate section here explaining how to disable SAML IdP, both on role and the cluster level?